src/namepath.rs

```diff
@@ -42,3 +63,11 @@ impl Serialize for Namepath<'_> {
     serializer.serialize_str(&format!("{self}"))
   }
 }
+
+impl<'src> Deref for Namepath<'src> {
```

Comment:
We shouldn't add `deref` to `Namepath`. It means that any function on `Vec` (I actually changed it to slice before I decided we shouldn't provide it at all.) is valid for namepath, which I think is not ideal. See comment about `last` later on.